src/org/labkey/test/components/ui/grids/GridFilterModal.java

```diff
@@ -0,0 +1,139 @@
+package org.labkey.test.components.ui.grids;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.UpdatingComponent;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.react.Tabs;
+import org.labkey.test.components.ui.search.FilterExpressionPanel;
+import org.labkey.test.components.ui.search.FilterFacetedPanel;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+/**
+ * Wraps 'labkey-ui-component' defined in <code>public/QueryModel/GridFilterModal.tsx</code>
+ */
+public class GridFilterModal extends ModalDialog
+{
+    protected static final Locator listItem = Locator.byClass("list-group-item");
+
+    private final UpdatingComponent _linkedComponent;
+
+    public GridFilterModal(WebDriver driver, UpdatingComponent linkedComponent)
+    {
+        super(new ModalDialogFinder(driver));
+        _linkedComponent = linkedComponent;
+    }
+
+    @Override
+    protected void waitForReady(ModalDialog.ElementCache ec)
+    {
+        getWrapper().shortWait().until(ExpectedConditions.visibilityOf(listItem.findElementOrNull(elementCache().fieldsSelectionPanel)));
+    }
```

Comment:
`visibilityOf` will NPE if `listItem.findElementOrNull` returns null. `findWhenNeeded` is what you want; it will play nicely with `visibilityOf` when the element isn't present.
```suggestion
    protected void waitForReady(ModalDialog.ElementCache ec)
    {
        getWrapper().shortWait().until(ExpectedConditions.visibilityOf(listItem.findWhenNeeded(elementCache().fieldsSelectionPanel)));
    }
```

Reply:
Done